### PR TITLE
Standardize tracing setup in rust

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
None. 

### What changes does this PR make to Grapl? Why?
Wanted to stop doing all this repeated stuff
image

This will lead into solving the "disable otel in prod" bug 
but now I can do it in one place not 13 :) 

### How were these changes tested?
CI

### CODE REVIEW

File: src/rust/generator-dispatcher/tests/test_utils/context.rs [hunk 1]
```diff
@@ -32,29 +24,7 @@ pub struct GeneratorDispatcherTestContext {
...
-        // register a subscriber
-        let filter = EnvFilter::from_default_env();
-        tracing_subscriber::registry()
-            .with(filter)
-            .with(log_layer)
-            .with(tracing_opentelemetry::layer().with_tracer(tracer))
-            .init();
-
-        tracing::info!("logger configured successfully");
+        let _guard = setup_tracing("generator-dispatcher-integration-tests").unwrap();
```
Comment (wimax-grapl) [inline src/rust/generator-dispatcher/tests/test_utils/context.rs:27]: the unwrap maps to the previous
>             .expect("could not configure tracer");

File: src/rust/generators/osquery-generator/src/main.rs [hunk 2]
```diff
@@ -10,45 +7,19 @@ use rust_proto::graplinc::grapl::{
...
 
 mod parsers;
 
 #[non_exhaustive]
 #[derive(Debug, Error)]
-pub enum OsqueryGeneratorError {
-    #[error("error configuring tracing {0}")]
-    TraceError(#[from] opentelemetry::trace::TraceError),
-}
+pub enum OsqueryGeneratorError {}
```
Comment (jgrillo-grapl) [inline src/rust/generators/osquery-generator/src/main.rs:15]: this doesn't appear to be used, and it's a little weird to have an export from `main.rs`

File: src/rust/generators/osquery-generator/src/main.rs [hunk 3]
```diff
@@ -10,45 +7,19 @@ use rust_proto::graplinc::grapl::{
...
-    let filter = EnvFilter::from_default_env();
-    tracing_subscriber::registry()
-        .with(filter)
-        .with(log_layer)
-        .with(tracing_opentelemetry::layer().with_tracer(tracer))
-        .init();
-
-    tracing::info!("logger configured successfully");
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let _guard = setup_tracing(SERVICE_NAME);
```
Comment (christophermaier) [inline src/rust/generators/osquery-generator/src/main.rs:22]: Why no `unwrap`/`expect` on this `setup_tracing` call?

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

File: src/rust/generators/osquery-generator/src/main.rs (hunk 2 above)
Comment (wimax-grapl) [reply src/rust/generators/osquery-generator/src/main.rs:15]: Sort of. This whole thing is a shambling corpse, but such an enum will be useful once ported to SDK

**Review by christophermaier — APPROVED** [review]
Nice simplification, modulo a few nagging consistency issues.

File: src/rust/node-identifier/src/main.rs [hunk 4]
```diff
@@ -43,30 +36,11 @@ use crate::{
...
-    // register a subscriber
-    let filter = EnvFilter::from_default_env();
-    tracing_subscriber::registry()
-        .with(filter)
-        .with(log_layer)
-        .with(tracing_opentelemetry::layer().with_tracer(tracer))
-        .init();
-
-    tracing::info!("logger configured successfully");
+    let _guard = setup_tracing(SERVICE_NAME)?;
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/node-identifier/src/main.rs:43]: Lots of the test situations are not -> Result, and as such, those I'm keeping as-is.

[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/node-identifier/src/main.rs:43]: AsyncTestContext doesnt support it for example

[the remark this answers is not in the document]
Comment (christophermaier) [reply src/rust/node-identifier/src/main.rs:43]: Can we use `?` for real code then, and `expect` in test code?

[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/node-identifier/src/main.rs:43]: yep, done. i'm actually mad at the rustl inter that it didn't enforce a must_use on these results

**Pull request merged**